auction-server/src/opportunity/service/make_permitted_tokens.rs

```diff
@@ -0,0 +1,53 @@
+use {
+    super::{
+        ChainTypeEvm,
+        Service,
+    },
+    crate::{
+        api::RestError,
+        opportunity::{
+            api::OpportunityBid,
+            contracts::TokenPermissions,
+            entities::opportunity_evm::OpportunityEvm,
+        },
+    },
+    ethers::types::U256,
+};
+
+pub struct MakePermittedTokensInput {
+    pub opportunity:     OpportunityEvm,
+    pub opportunity_bid: OpportunityBid,
+}
+
+impl Service<ChainTypeEvm> {
+    pub fn make_permitted_tokens(
```

Comment:
I think this separation is a bit too extreme. `make_permitted_tokens` is just some internal function that is used by another function. We shouldn't expose it publicly and I'd rather have this next to the caller. We can discuss this but maybe each file in this folder should include the logic for a complete functionality this service exposes to outside.

Reply:
We should make private some of the services functionalities